pythran/pythonic/os/path/join.hpp

```diff
@@ -32,11 +38,15 @@ namespace pythonic
       void _join(types::str &buffer, T &&head, Types &&... tail)
       {
         if (head[0] == '/')
```

Comment:
why is this char still here and not turned into another check? BTW, in addition of #defining OS_SEP, you should add a function is_path_sep that takes into account the fact that windows can use both / and \ as path sep.